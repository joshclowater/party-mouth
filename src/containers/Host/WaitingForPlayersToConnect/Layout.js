import React from 'react';
import styled from 'styled-components';
import Button from 'src/components/Button';
import { startGame } from 'src/socket';

const Wrapper = styled.div`
  text-align: center;
  height: 60vh;
`;

const GameInfo = styled.div`
  height: 25vh;
`;

const GameId = styled.div`
  font-size: 8vh;
  margin-bottom: 3vh;
`;

const Players = styled.div`
  margin-bottom: 4vh;
`;

const PlayersHeader = styled.div`
  font-size: 4vh;
  margin-bottom: 2vh;
`;

const Player = styled.div`
  margin-bottom: 1.5vh;
`;

export default function(props) {
  return (
    <Wrapper>
      <GameInfo>
        <div>Your game id is</div>
        <GameId id="gameId">{props.gameId}</GameId>
        <div>Enter the id on additional devices to connect</div>
      </GameInfo>
      {props.players.length > 0 && (
        <Players id="players">
          <PlayersHeader>Players</PlayersHeader>
          {props.players.map(player => (
            <Player key={player.id}>{player.name}</Player>
          ))}
        </Players>
      )}
      {props.players.length >= 4 && (
        <Button id="startGame" onClick={startGame}>
          Start game
        </Button>
      )}
    </Wrapper>
  );
}
