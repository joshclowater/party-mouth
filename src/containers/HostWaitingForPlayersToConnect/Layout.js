import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  text-align: center;
  height: 60vh;
`;

const GameInfo = styled.div`
  height: 30vh;
`;

const GameId = styled.div`
  font-size: 8vh;
  margin-bottom: 3vh;
`;

const PlayersHeader = styled.div`
  font-size: 4vh;
  margin-bottom: 2vh;
`;

const Player = styled.div`
  margin-bottom: 1.5vh;
`;

export default function(props) {
  return (
    <Wrapper>
      <GameInfo>
        <div>Your game id is</div>
        <GameId id="gameId">{props.gameId}</GameId>
        <div>Enter the id on additional devices to connect</div>
      </GameInfo>
      {props.players.size > 0 && (
        <div id="players">
          <PlayersHeader>Players</PlayersHeader>
          {props.players.map(player => (
            <Player key={player.get('name')}>{player.get('name')}</Player>
          ))}
        </div>
      )}
    </Wrapper>
  );
}
