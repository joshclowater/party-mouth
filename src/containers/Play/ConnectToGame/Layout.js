import React from 'react';
import styled from 'styled-components';

import Button from 'src/components/Button';
import InputField from 'src/components/InputField';

const Form = styled.form`
  width: 50vw;
  min-width: 200px;
`;

export default function PlayerConnectToGame(props) {
  return (
    <Form onSubmit={props.handleSubmit}>
      <InputField
        name="gameId"
        label="Game id"
        placeholder="Enter a 5-letter game id"
        autoComplete="off"
        maxLength="5"
      />
      <InputField
        name="name"
        label="Your name (max length 12)"
        placeholder="Enter your name"
        autoComplete="given-name"
      />
      <Button id="joinGame" type="submit" disabled={props.submitting}>
        {props.submitting ? 'Connecting...' : 'Join game'}
      </Button>
    </Form>
  );
}
