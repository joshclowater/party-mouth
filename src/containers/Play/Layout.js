import React, { Component } from 'react';
import styled from 'styled-components';
import { initSocket } from 'src/socket';

import CenteredComponent from 'src/components/Centered';
import ConnectingToServer from 'src/components/ConnectingToServer';
import PlayerConnectToGame from 'src/containers/PlayerConnectToGame';

const CenteredScreen = styled(CenteredComponent)`
  height: 100vh;
`;

export default class Play extends Component {
  componentDidMount() {
    initSocket('player');
  }

  render() {
    let content;
    if (this.props.status === 'connecting-to-server') {
      content = <ConnectingToServer />;
    } else if (this.props.status === 'waiting-to-join') {
      content = <PlayerConnectToGame />;
    } else if (this.props.status === 'waiting-for-game-start') {
      content = 'Connected. Waiting for game to start.';
    } else {
      throw new Error('invalid status', this.props.status);
    }
    return <CenteredScreen>{content}</CenteredScreen>;
  }
}
