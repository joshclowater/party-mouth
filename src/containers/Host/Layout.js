import React, { Component } from 'react';
import styled from 'styled-components';
import { initSocket } from 'src/socket';

import HostWaitingForPlayersToConnect from 'src/containers/HostWaitingForPlayersToConnect';
import CenteredComponent from 'src/components/Centered';
import ConnectingToServer from 'src/components/ConnectingToServer';

const CenteredScreen = styled(CenteredComponent)`
  height: 100vh;
`;

export default class Host extends Component {
  componentDidMount() {
    initSocket('host');
  }

  render() {
    let content;
    if (this.props.status === 'connecting-to-server') {
      content = <ConnectingToServer />;
    } else if (this.props.status === 'waiting-for-players-to-connect') {
      content = <HostWaitingForPlayersToConnect />;
    } else {
      throw new Error('invalid status', this.props.status);
    }
    return <CenteredScreen>{content}</CenteredScreen>;
  }
}
