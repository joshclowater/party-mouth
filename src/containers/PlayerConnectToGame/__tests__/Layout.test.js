import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../Layout';

it('Renders correctly', () => {
  expect(
    shallow(<Layout submitting="mock submitting value" />)
  ).toMatchSnapshot();
});

describe('submitting is true', () => {
  const wrapper = shallow(<Layout submitting />);
  it('Renders "Connecting..." in the button', () => {
    expect(wrapper.find('#joinGame').get(0).props.children).toEqual(
      'Connecting...'
    );
  });
});

describe('submitting is false', () => {
  const wrapper = shallow(<Layout submitting={false} />);
  it('Renders "Join game" in the button', () => {
    expect(wrapper.find('#joinGame').get(0).props.children).toEqual(
      'Join game'
    );
  });
});
