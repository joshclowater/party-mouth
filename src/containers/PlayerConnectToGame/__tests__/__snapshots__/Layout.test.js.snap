// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<styled.form>
  <InputField
    autoComplete="off"
    label="Game id"
    maxLength="5"
    name="gameId"
    placeholder="Enter a 5-letter game id"
  />
  <InputField
    autoComplete="given-name"
    label="Your name (max length 12)"
    name="name"
    placeholder="Enter your name"
  />
  <styled.button
    disabled="mock submitting value"
    id="joinGame"
    type="submit"
  >
    Connecting...
  </styled.button>
</styled.form>
`;
